### PR TITLE
Add ZLib compression test data

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @carlossanlop @ericstj
Contributes to https://github.com/dotnet/runtime/issues/2236

### CODE REVIEW

File: src/System.IO.Compression.TestData/ZLibTestData/cp.html.z [hunk 1]
```diff
@@ -0,0 +1,646 @@
+x`�<head>
+<title>Compression Pointers</title>
```
Comment (ericstj) [inline src/System.IO.Compression.TestData/ZLibTestData/cp.html.z:2]: Shouldn’t this file be compressed?

File: src/System.IO.Compression.TestData/ZLibTestData/xargs.1.z [hunk 2]
```diff
@@ -0,0 +1,113 @@
+x�|�.TH XARGS 1L \" -*- nroff -*-
```
Comment (ericstj) [inline src/System.IO.Compression.TestData/ZLibTestData/xargs.1.z:1]: Shouldn’t this file be compressed?

File: src/System.IO.Compression.TestData/ZLibTestData/cp.html.z (hunk 1 above)
Comment (stephentoub) [reply src/System.IO.Compression.TestData/ZLibTestData/cp.html.z:2]: I used NoCompression for some, Fastest for some, and Optimal for some.

Comment (ericstj) [reply src/System.IO.Compression.TestData/ZLibTestData/cp.html.z:2]: I see.  Mix is fine, I honestly don't know what was done for the other data here, likely it was all compressed at optimal level.  cc @ianhays

Comment (stephentoub) [reply src/System.IO.Compression.TestData/ZLibTestData/cp.html.z:2]: Yeah, I was just guessing here :-)

Comment (ericstj) [reply src/System.IO.Compression.TestData/ZLibTestData/cp.html.z:2]: Perhaps we can fix that by adding a readme that indicates how the files were generated and how they can be regenerated?

**Review by ericstj — APPROVED** [review]
LGTM modulo suggestion to document how these were created.

File: src/System.IO.Compression.TestData/ZLibTestData/cp.html.z (hunk 1 above)
Comment (stephentoub) [reply src/System.IO.Compression.TestData/ZLibTestData/cp.html.z:2]: Sure, added a README.md with the code for a simple C# program to do it.

**Pull request merged**